src/module/system/text-editor.ts

```diff
@@ -553,34 +553,26 @@ class TextEditorPF2e extends TextEditor {
               ? "all"
               : "gm";
 
-        const type = rawParams.type ?? "check";
+        const type = rawParams.type.trim();
+        const overideTraits =
+            typeof rawParams.overrideTraits === "string" && ["true", ""].includes(rawParams.overrideTraits);
+        const rawTraits = rawParams.traits?.split(",").map((t) => t.trim()) ?? [];
+        const traits = R.uniq(R.compact(overideTraits ? rawTraits : [rawTraits, item?.system.traits.value].flat()));
+
         const params: CheckLinkParams = {
             ...rawParams,
             type,
-            basic: "basic" in rawParams,
+            basic: typeof rawParams.basic === "string" && ["true", ""].includes(rawParams.basic),
+            dc: rawParams.dc?.trim() || null,
+            defense: rawParams.defense?.trim() || null,
             showDC,
-            traits: (() => {
-                const traits: string[] = [];
-                // Set item traits
-                const itemTraits = item?.system.traits?.value ?? [];
-                if (rawParams.overrideTraits !== "true") {
-                    traits.push(...itemTraits);
-                }
-
-                // Add param traits
-                if (rawParams.traits) traits.push(...rawParams.traits.split(",").map((trait) => trait.trim()));
-
-                // Deduplicate traits
-                return Array.from(new Set(traits));
-            })(),
+            traits,
             // Set action slug, damaging effect for basic saves, and any parameterized options
             extraRollOptions: R.compact([
-                ...(rawParams.basic === "true" ? ["damaging-effect"] : []),
+                ...("basic" in rawParams ? ["damaging-effect"] : []),
```

Comment:
```suggestion
                ...(params.basic ? ["damaging-effect"] : []),
```

I wonder why it didn't do this from the start

Reply:
data entry code